src/client/eppo-client.ts

```diff
@@ -331,20 +350,54 @@ export default class EppoClient implements IEppoClient {
 
     // Compute variation for subject.
     const { subjectShards } = experimentConfig;
-    const { variations } = allocation;
+    const { variations, holdout } = allocation;
 
-    const shard = getShard(`assignment-${subjectKey}-${flagKey}`, subjectShards);
-    const assignedVariation = variations.find((variation) =>
-      isShardInRange(shard, variation.shardRange),
-    );
+    let assignedVariation;
+    let holdoutKey = null;
+    let holdoutVariation = null;
 
-    const internalAssignment = {
+    if (holdout && this.isInHoldoutSample(subjectKey, flagKey, experimentConfig, holdout)) {
```

Comment:
Checking if holdout is `null`, which it would be if there is currently no active holdout